test/postgresql.test.js

```diff
@@ -384,18 +384,76 @@ describe('postgresql connector', function() {
         });
     });
 
+    it('supports like for date types (with explicit typecast)', function(done) {
+      PostWithDate.find({where: {created: {like: '%05%'}}}, function(err, result) {
+        should.not.exists(err);
+        result.length.should.equal(1);
+        done();
+      });
+    });
+
+    it('supports case insensitive queries using ilike for date types (with explicit typecast)', function(done) {
+      PostWithDate.find({where: {created: {ilike: '%05%'}}}, function(err, result) {
+        should.not.exists(err);
+        result.length.should.equal(1);
+        done();
+      });
+    });
+
+    it('supports nlike for no match for date types (with explicit typecast)', function(done) {
+      PostWithDate.find({where: {content: {nlike: '%07%'}}},
+        function(err, posts) {
+          should.not.exists(err);
+          posts.length.should.equal(2);
+          done();
+        });
+    });
+
+    it('supports case insensitive queries using nilike for date types (with explicit typecast)', function(done) {
+      PostWithDate.find({where: {created: {nilike: '%07%'}}}, function(err, result) {
+        should.not.exists(err);
+        result.length.should.equal(2);
+        done();
+      });
+    });
+
     function deleteTestFixtures(done) {
       Post.destroyAll(done);
     }
 
     function createTestFixtures(done) {
-      Post.create([{
-        title: 't1',
-        content: 'T1_TestCase',
-      }, {
-        title: 't2',
-        content: 'T2_TheOtherCase',
-      }], done);
+      db.automigrate(function(err, result) {
+        if (err) throw err;
+        PostWithDate = db.define('PostWithDate', {
+              title: {type: String, length: 255, index: true},
+              content: {type: String},
+              created: {
+                type: String,
+                postgresql: {
+                  dataType: 'timestamp with time zone'
+                }
+              }
+            });
+        Post.create([{
+          title: 't1',
+          content: 'T1_TestCase',
+        }, {
+          title: 't2',
+          content: 'T2_TheOtherCase',
+        }]);
+        PostWithDate.create([
```

Comment:
@ptrblgh There is an issue here. Since the functions are async, it does not work like that and hence the tests are failing. You need to do something like this:
```
Post.create([...], function(err, result) { should.not.exist(err); PostWithDate.create([..], done});
```